docs/JSON_format.md

```diff
@@ -634,14 +634,14 @@ attributes for these objects are:
     - `action_id`: the id of the default action
     - `action_const`: an optional boolean value which is `true` iff the control
     plane is not allowed to change the default action function. Default value is
-    `false`. It can only be set to `true` for `simple` tables.
+    `false`. It can only be set to `true` for `simple` tables.  See Note 2 below.
     - `action_data`: an optional JSON array where each entry is the hexstring
```

Comment:
is this attribute (`action_data`) always set by p4c nowadays?

Reply:
I committed some additional checks that threw exceptions if `action_id` or `action_data` were ever absent inside of a `default_entry` object value, and they were always found for all p4c tests, so it seems very likely that they are guaranteed to be present.  `default_entry` is always present for tables with type `simple`, too.

Reply:
The change to throw an error if `default_entry` is not present in a table with type `simple` led to updating more test cases than I expected.  I can back out those changes if they seem like busy-work.

Reply:
I don't think we should change these old test cases, it just creates noise. However, I think you should mention it in the Node 2 below.

Reply:
Just so I am clear on your meaning, your recommendation is:
(a) document that `default_entry` is included in all simple tables in today's p4c
(b) remove the check that throws an exception if `default_entry` is not present in a simple table
(c) remove all changes to old tests
Yes?

Reply:
I have removed the code that threw an error if `default_entry` was absent in a table definition, restoring it back to optional, as it was before this commit.  I also removed all the commits in the old unit test cases that were needed to make those old unit tests pass with such a change.

Reply:
> I don't think we should change these old test cases, it just creates noise. However, I think you should mention it in the Node 2 below.

I have added to Note 2 in the JSON docs a mention that p4c always include key `default_entry` for tables with `type` equal to `simple`, and always omits `default_entry` for other types of tables.